tsl/src/partialize_finalize.c

```diff
@@ -518,21 +518,29 @@ tsl_finalize_agg_sfunc(PG_FUNCTION_ARGS)
 	PG_RETURN_POINTER(tstate);
 }
 
-/* tsl_finalize_agg_ffunc:
- * apply the finalize function on the state we have accumulated
+/*
+ * Apply the finalize function on the state we have accumulated
+ *
+ * The state passed down to this function can be used by other aggregate
+ * functions so it is important to not change the state when computing the
+ * final result.
```

Comment:
Didn't know about this. Is it some optimization like using the same state for avg, count and sum?

Reply:
I think the comment should say something in the spirit of "the partial state should not be modified by the finalization function because it can be reused". I thought the same thing about avg and sum but apparently the reproducer only has 1 aggregate function.

Reply:
> Didn't know about this. Is it some optimization like using the same state for avg, count and sum?

It seems to be mainly used in hash aggregation, but I see it pop up in group aggregation as well. It is the `finalize_aggregates` function in `nodeAgg.c` that is calling this using the hashed aggregates.